esmvalcore/_task.py

```diff
@@ -290,28 +287,22 @@ def _initialize_cmd(self):
             raise DiagnosticError(f"{err_msg}: file does not exist.")
 
         cmd = []
-        if not os.access(script_file, os.X_OK):  # if not executable
-            interpreters = {
-                'jl': 'julia',
-                'ncl': 'ncl',
-                'py': 'python',
-                'r': 'Rscript',
-            }
-            args = {
-                'ncl': ['-n', '-p'],
-            }
-            if self.settings['profile_diagnostic']:
-                profile_file = Path(self.settings['run_dir']) / 'profile.json'
-                args['py'] = [
-                    '-m', 'vprof', '-o',
-                    str(profile_file), '-c', 'c'
-                ]
 
-            ext = script_file.suffix.lower()[1:]
-            if ext not in interpreters:
-                raise DiagnosticError(
-                    f"{err_msg}: non-executable file with unknown extension "
-                    f"'{script_file.suffix}'.")
+        interpreters = {
+            'jl': 'julia',
+            'ncl': 'ncl',
+            'py': 'python',
+            'r': 'Rscript',
```

Comment:
I wonder if you can do something like this here:

```suggestion
            interpreters = {
                'jl': which('julia'),
                'ncl': which('ncl'),
                'py': sys.executable,
                'r': which('Rscript'),
            }
```

Reply:
Not quite, because [sys.executable](https://docs.python.org/3/library/sys.html#sys.executable) can also be `None` apparently, they then we would end up with
```suggestion
            'jl': 'julia',
            'ncl': 'ncl',
            'py': sys.executable if sys.executable else 'python',
            'r': 'Rscript',
```
Do you think that is still better?

Reply:
I wonder in which cases it would be `None`... Anyway, it is fine the way it is, I was just wondering if this would be a simpler way to do it, but I didn't directly see it 😅

Reply:
OK, I'll leave it as it is then.